Mathlib/RingTheory/Localization/FieldRange.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.RingTheory.Localization.FractionRing
+import Mathlib.FieldTheory.Adjoin
+
+/-!
+
+# Some results on image of a ring and its fraction field under ring homomorphisms
+
+-/
+
+namespace IsFractionRing
+
+variable {F A K L : Type*} [Field F] [CommRing A] [Algebra F A]
+  [Field K] [Algebra F K] [Algebra A K] [IsFractionRing A K] [Field L] [Algebra F L]
+
+section RingHom
+
+variable {g : A →+* L} {f : K →+* L}
+
+/-- If `A` is a ring with fraction field `K`, `L` is a field, `g : A →+* L` lifts to `f : K →+* L`,
+then the image of `f` is the field genereated by the image of `g`. -/
+theorem ringHom_fieldRange_eq_of_comp_eq
+    (h : RingHom.comp f (algebraMap A K) = g) :
+    f.fieldRange = Subfield.closure g.range := by
+  refine le_antisymm ?_ (by simp [← h, Set.range_comp_subset_range])
+  rintro x ⟨y, rfl⟩
+  obtain ⟨_, _, -, rfl⟩ := div_surjective (A := A) y
+  simp_rw [map_div₀, ← RingHom.comp_apply, h]
+  apply Subfield.div_mem <;> (apply Subfield.subset_closure; simp)
```

Comment:
Continued from https://github.com/leanprover-community/mathlib4/pull/18648#discussion_r1838586450:
I mean that if you add
```
theorem closure_range_algebraMap : Subfield.closure (Set.range (algebraMap A K)) = ⊤ :=
  top_unique fun z _ ↦ by
    obtain ⟨x, y, hy, rfl⟩ := div_surjective (A := A) z
    apply div_mem <;> exact Subfield.subset_closure ⟨_, rfl⟩
```
after `div_surjective` (need to import Subfield.Basic, +2 transitive imports) then the following works:
```suggestion
theorem ringHom_fieldRange_eq_of_comp_eq
    (h : RingHom.comp f (algebraMap A K) = g) :
    f.fieldRange = Subfield.closure g.range := by
  rw [f.fieldRange_eq_map, ← IsFractionRing.closure_range_algebraMap (A := A),
    f.map_field_closure, ← Set.range_comp, ← f.coe_comp, h, g.coe_range]
```

Reply:
Thank you very much! I'll study it next day.

Reply:
Good observation! Added.